### PR TITLE
Add NSTC and NRWC normalization

### LINKED ISSUE #120: normalization of timeseries
I would pledge for using NREF normalization for comparing SI and CPV and  NREA normalization for comparing all technologies, especially persosi but also taking into account that CPV is not fully developed yet 

Extract from our rtd:
For the energy system optimization normalized timeseries are needed, which can then be scaled to the optimal installation size of the system.

There is three different ways to normalize the PV timeseries.

1) **Normalize by peak power (NP)**

* This procedure takes into account the impacts of the weather on the profile, but not on the scaling of the timeseries

2) **Normalize by reference p_mp (power at maximum powerpoint) (NREF)**

* This procedure accounts for all losses under real world conditions and displays the difference between ideal operation and real world operation

3) **Normalize by realworld p_mp (NREA)**

* This procedure calculates the maximum powerpoint for realorld conditions at irr_ref = 1000 W/qm and temp_ref = 25 °.

* This way it treats the technology as if it was "ideal" under real world conditions.
* This normalization is of great importance when it comes to estimating technologies that are still under development and do not reach their reference p_mp yet.

### PR DESCRIPTION
**Pull request by Piranias**
Fix #120

**Changes proposed in this pull request**:
- adding a function "get_peak" to calculate either NRWC or NSTC peak value
- adding a function "calculate_NRWC_peak() to calculate NRWC pea value from weather data (Madrid, Spain, 2014)

The following steps were realized, as well (if applies):
- [x] Use in-line comments to explain your code
- [x] Write docstrings to your code
- [x] For new functionalities: Explain in readthedocs -> see other PR
- [x] Write test(s) for your new patch of code
- :x: Update the CHANGELOG.md (let's start this after the first release)
- [x] Apply black (`black . --exclude docs/`)

*Please mark above checkboxes as following:*
- [ ] Open
- [x] Done

:x: Check not applicable to this PR

<sub>*For more information on how to contribute check the [CONTRIBUTING.md](https://github.com/rl-institut/mvs_eland/blob/dev/CONTRIBUTING.md).*<sub>

### CODE REVIEW

File: pvcompare/main.py [hunk 1]
```diff
@@ -84,7 +84,7 @@ def main(
         input_directory=input_directory,
         mvs_input_directory=mvs_input_directory,
         year=year,
-        normalized=True,
+        normalization="NREA",
```
Comment (SabineHaas) [inline pvcompare/main.py:87]: below  `normalization="NRWC"` is set - is `normalization` of main() given to another function?

File: pvcompare/pv_feedin.py [hunk 2]
```diff
@@ -86,6 +86,14 @@ def create_pv_components(
         if None: ./data/mvs_inputs/
     directory_energy_production: str
         if None: ./data/mvs_inputs/elements/csv/
+    psi_type: str
+        "Korte" or "Chen"
+    normalization: str
+        "NP": Normalize by peak power
+        "NSTC": Normalize by reference p_mp
+        "NRWCL": Normalize by realworld p_mp
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:94]:
```suggestion
        "NRWC": Normalize by realworld p_mp
```

File: pvcompare/pv_feedin.py [hunk 3]
```diff
@@ -360,7 +371,11 @@ def create_si_time_series(lat, lon, weather, surface_azimuth, surface_tilt, norm
         surface azimuth of the modules
     surface_tilt: float
         surface tilt of the modules
-    normalized: boolean
+    normalization: str
+        "NP": Normalize by peak power
+        "NSTC": Normalize by reference p_mp
+        "NRWCL": Normalize by realworld p_mp
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:377]:
```suggestion
        "NRWC": Normalize by realworld p_mp
```

File: pvcompare/pv_feedin.py [hunk 4]
```diff
@@ -419,54 +437,66 @@ def create_cpv_time_series(
         Surface azimuth of the modules (180° for south, 270° for west, etc.).
     surface_tilt: float
         Surface tilt of the modules. (horizontal=90° and vertical=0°)
-    normalized: bool
-        If True, the time series is divided by the peak power of the CPV
-        module. Default: False.
+    normalization: str
+        "NP": Normalize by peak power
+        "NSTC": Normalize by reference p_mp
+        "NRWCL": Normalize by realworld p_mp
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:443]:
```suggestion
        "NRWC": Normalize by realworld p_mp
```

File: pvcompare/pv_feedin.py [hunk 5]
```diff
@@ -419,54 +437,66 @@ def create_cpv_time_series(
...
                 lat, lon, weather, surface_azimuth, surface_tilt
             )
-            / peak
-        ).clip(0)
+            / 1000
+        )
+
     else:
-        logging.info("Absolute CPV time series is calculated in kW.")
+        logging.info("Normalized CPV time series is calculated in kW.")
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:467]: Is the unit here "kW/kWp", depending on the normalization method? Or just no unit?

File: pvcompare/pv_feedin.py [hunk 6]
```diff
@@ -485,9 +515,11 @@ def create_psi_time_series(
     psi_type  : str
         Defines the type of module of which the time series is calculated.
         Options: "Korte", "Chen"
-    normalized: bool
-        If True, the time series is divided by the peak power of the CPV
-        module. Default: False.
+    normalization: str
+        "NP": Normalize by peak power
+        "NSTC": Normalize by reference p_mp
+        "NRWCL": Normalize by realworld p_mp
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:521]:
```suggestion
        "NRWC": Normalize by realworld p_mp
```

File: pvcompare/pv_feedin.py [hunk 7]
```diff
@@ -509,22 +542,27 @@ def create_psi_time_series(
...
                 atmos_data=atmos_data,
-                number_hours=8760,
+                number_hours=number_rows,
                 psi_type=psi_type,
             )
             / 1000
         )
     else:
-        logging.info("Normalized PSI time series is calculated.")
+        logging.info("Normalized PSI time series is calculated in kW.")
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:551]: unit, as mentioned above

File: pvcompare/pv_feedin.py [hunk 8]
```diff
@@ -610,59 +661,153 @@ def nominal_values_pv(technology, area, surface_azimuth, surface_tilt, psi_type)
...
-    )
+def get_peak(technology, normalization, module_parameters_1, module_parameters_2):
+    """
+    this function returns the peak value for the given technology and the given
+    type of normalization.
+
+    :param technology: str
+        "si", "cpv" or "psi"
+    :param normalization: str
+        kind of normalization
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:672]: Could you add the options as above, please?

File: pvcompare/pv_feedin.py [hunk 9]
```diff
@@ -610,59 +661,153 @@ def nominal_values_pv(technology, area, surface_azimuth, surface_tilt, psi_type)
...
+        return calculate_NRWC_peak(technology=technology)
+
+
+def calculate_NRWC_peak(technology):
+    """
+    calculates the peak value of a technology under real world conditions.
+
+    Using weather year of Madrid, Spain, 2014 the dataframe is filtered to find
+    the timestep where irradiance and temperature come closest to reference
+    conditions of ghi=1000 W/m and temp_air= 25 °C. #todo: cell temperature instead of air temperature?
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:710]: cell temperature, right?

File: pvcompare/pv_feedin.py [hunk 10]
```diff
@@ -610,59 +661,153 @@ def nominal_values_pv(technology, area, surface_azimuth, surface_tilt, psi_type)
...
-        surface_azimuth=surface_azimuth,
+    input_directory = constants.DEFAULT_INPUT_DIRECTORY
+    weather_file = os.path.join(
+        input_directory, "weatherdata_52.52437_13.41053_2014.csv"
+    )
+    if os.path.isfile(weather_file):
+        weather = pd.read_csv(weather_file, index_col=0)
+    else:
+        logging.error(
+            f"the weather file {weather_file} does not exist. Please"
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:734]: Does this file come with pvcompare?

File: pvcompare/pv_feedin.py [hunk 11]
```diff
@@ -610,59 +661,153 @@ def nominal_values_pv(technology, area, surface_azimuth, surface_tilt, psi_type)
...
+    # check if cell temperature = temp_ref
+    weather["cell_temperature"] = pvlib.temperature.pvsyst_cell(
+        poa_global=weather["poa_global"],
+        temp_air=weather["temp_air"],
+        wind_speed=weather["wind_speed"],
+    )
+    cell_temp = weather["cell_temperature"]
+    irrad = weather["poa_global"]
+
+    peak_irr = weather.iloc[(weather["poa_global"] - irr_ref).abs().argsort()[:2]]
```
Comment (SabineHaas) [inline pvcompare/pv_feedin.py:763]: Could you add a comment here? You select the hour with the data closest to STC?

File: tests/test_pv_feedin.py [hunk 12]
```diff
@@ -55,25 +72,36 @@ def test_create_si_times_eries(self):
             weather=self.weather,
             surface_azimuth=self.surface_azimuth,
             surface_tilt=self.surface_tilt,
-            normalized=False,
+            normalization=None,
         )
         output = round(ts.values.sum(), 3)
-        assert output == 0.113
+        assert output == 0.216
```
Comment (SabineHaas) [inline tests/test_pv_feedin.py:78]: Did you check why the output has changed?

Comment (SabineHaas) [reply tests/test_pv_feedin.py:78]: Ah you've changed the conditions above - why have you changed them?

**Review by SabineHaas — APPROVED** [review]
Nice feature! I didn't have time to go through the tests, which is why I asked questions.

File: pvcompare/pv_feedin.py (hunk 5 above)
Comment (Piranias) [reply pvcompare/pv_feedin.py:467]: yes, if you keep in mind that there is two different "peak" definition that that's correct

File: pvcompare/pv_feedin.py (hunk 10 above)
Comment (Piranias) [reply pvcompare/pv_feedin.py:734]: yes

File: tests/test_pv_feedin.py (hunk 12 above)
Comment (Piranias) [reply tests/test_pv_feedin.py:78]: I tried out to set the cell temperature to 0, but then decided against it

File: pvcompare/pv_feedin.py (hunk 5 above)
Comment (Piranias) [reply pvcompare/pv_feedin.py]: I added this

File: pvcompare/main.py (hunk 1 above)
Comment (Piranias) [reply pvcompare/main.py]: fixed this

**Pull request merged**